### ISSUE TITLE
burnin_epochs in MVEBase not used

### ISSUE THREAD

**Issue by codingS3b** [lightning-uq-box/lightning-uq-box#150]
It seems the `MVEBase` class has an unused parameter `burnin_epochs`. I mean, it is used in the `training_step` later on, but is not touched in the constructor.
In the derived class  `MVERegression` this parameter is passed to the base constructor but is not used there and only logged
to the hyperparameters in the derived class which confused me a bit. 
So maybe removing this parameter from the base class could be an option or calling the `self.save_hyperparameters` already in the base class?

**Comment by nilsleh** [comment]
Good point. I think, making it a class variable in the Base class makes sense. I previously encountered some issues with lightning, when calling `self.save_hyperparameters` in a base constructor and not the derived class. So I think adding `self.burin_epochs=burnin_epochs` to `MVEBase` and accessing that instead of `self.hparams.burnin_epochs` would make things more clear?

**Comment by codingS3b** [comment]
That's also a good strategy. Then, maybe `burnin_epochs` should go to the `ignore` list in the call to `save_hyperparameters` to avoid having the same info twice, maybe? Or do you think it is still a good idea to log it there?

**Comment by nilsleh** [comment]
I think it does not need to be ignored since it is just a `int` type. The things included in the ingore are generally `nn.Modules` because they are already stored with the lightning module, and you get the suggestion to ignore them if you don't because then the more memory intensive variables are stored twice.

**Comment by codingS3b** [comment]
Another point I just noticed regarding the `burnin_epochs` parameter: It is lacking in the `MVEPxRegression`, apparently because it is derived from a different base class. 

Nevertheless, some initial epochs, where just the mean and not the sigma gets optimized could possibly be also useful in this case, or was there a particular reason to not implement it there?

**Comment by nilsleh** [comment]
Yes, you are correct, that is a bug. It kind of goes back to a design question about inheritance, where you have method specific functions, and functions that are specific to the kind of task (classification, px regression, etc.) and in this case I suppose I missed having the correct training step from MVE. If you want you can also include it in this PR?

The whole design question of specific task and UQ functions has introduced some code duplication, that more experienced python people could probably avoid :)
